Sources/MapboxDirections/Incident.swift

```diff
@@ -21,46 +20,174 @@ public struct Incident: Codable, Equatable {
-        self.geometryIndexStart = geometryIndexStart
-        self.geometryIndexEnd = geometryIndexEnd
+        self.rawLanesBlocked = Set(lanesBlocked.map { $0.rawValue })
+        self.shapeIndexRange = shapeIndexRange
+    }
+    
+    public init(from decoder: Decoder) throws {
+        let container = try decoder.container(keyedBy: CodingKeys.self)
+        let formatter = ISO8601DateFormatter()
+        
+        identifier = try container.decode(String.self, forKey: .identifier)
+        rawKind = try container.decode(String.self, forKey: .type)
+        kind = Kind(rawValue: rawKind)
+        
+        description = try container.decode(String.self, forKey: .description)
+        
+        if let date = formatter.date(from: try container.decode(String.self, forKey: .creationDate)) {
+            creationDate = date
+        } else {
+            throw DecodingError.dataCorruptedError(forKey: .creationDate,
+                                                   in: container,
+                                                   debugDescription: "`Intersection.creationTime` is encoded with invalid format.")
+        }
+        if let date = formatter.date(from: try container.decode(String.self, forKey: .startDate)) {
+            startDate = date
+        } else {
+            throw DecodingError.dataCorruptedError(forKey: .startDate,
+                                                   in: container,
+                                                   debugDescription: "`Intersection.startTime` is encoded with invalid format.")
+        }
+        if let date = formatter.date(from: try container.decode(String.self, forKey: .endDate)) {
+            endDate = date
+        } else {
+            throw DecodingError.dataCorruptedError(forKey: .endDate,
+                                                   in: container,
+                                                   debugDescription: "`Intersection.endTime` is encoded with invalid format.")
+        }
+        
+        impact = try container.decodeIfPresent(String.self, forKey: .impact)
+        subtype = try container.decodeIfPresent(String.self, forKey: .subtype)
+        subtypeDescription = try container.decodeIfPresent(String.self, forKey: .subtypeDescription)
+        alertCodes = try container.decode(Set<Int>.self, forKey: .alertCodes)
+        
+        rawLanesBlocked = try container.decode(Set<String>.self, forKey: .lanesBlocked)
+        lanesBlocked = rawLanesBlocked.reduce(into: Set<BlockedLane?>()) { $0.insert(BlockedLane(rawValue: $1)) }
+        
+        let geometryIndexStart = try container.decode(Int.self, forKey: .geometryIndexStart)
+        let geometryIndexEnd = try container.decode(Int.self, forKey: .geometryIndexEnd)
+        shapeIndexRange = geometryIndexStart..<geometryIndexEnd
+    }
+    
+    public func encode(to encoder: Encoder) throws {
+        var container = encoder.container(keyedBy: CodingKeys.self)
+        let formatter = ISO8601DateFormatter()
```

Comment:
ISO8601DateFormatter outputs a date in RFC&nbsp;3339 format. Would MapboxNavigationNative be OK with that format, or does it need to be a different ISO&nbsp;8601 format?

/ref mapbox/mapbox-navigation-ios#2327

Reply:
Oh, I wanted to avoid introducing custom extension, but didn't know about such issue.

Reply:
`Incident` dates are originally supplied in `yyyy-MM-dd'T'HH:mm:ssZ` format and NN takes that while setting a Route, so I don't think there's a reason to change it.